filters/auth/jwt_validation.go

```diff
@@ -0,0 +1,301 @@
+// configuration and check used.
+func (f *jwtValidationFilter) String() string {
+	return fmt.Sprintf("%s(%s)", JwtValidationName, strings.Join(f.claims, ","))
+}
+
+func (f *jwtValidationFilter) validateAnyClaims(token jwt.Token) bool {
+	for _, wantedClaim := range f.claims {
+		if claims, ok := token.Claims.(jwt.MapClaims); ok {
+			if _, ok2 := claims[wantedClaim]; ok2 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (f *jwtValidationFilter) Request(ctx filters.FilterContext) {
+	r := ctx.Request()
+
+	var info jwt.Token
+	infoTemp, ok := ctx.StateBag()[oidcClaimsCacheKey]
+	if !ok {
+		token, ok := getToken(r)
+		if !ok || token == "" {
+			unauthorized(ctx, "", missingToken, "", "")
+			return
+		}
+
+		parsedToken, err := jwt.Parse(token, func(token *jwt.Token) (interface{}, error) {
+			kid := token.Header["kid"].(string)
+
+			m.Lock()
+			defer m.Unlock()
+
+			if rsakeys[kid] == nil {
+				return nil, fmt.Errorf("not able to find public key for kid %s", kid)
+			}
+
+			return rsakeys[kid], nil
+		})
+		if err != nil {
+			log.Errorf("Error while parsing jwt token : %v.", err)
+			unauthorized(ctx, "", "error parsing jwt token", "", "")
+			return
+		} else if !parsedToken.Valid {
+			log.Errorf("Invalid token")
+			unauthorized(ctx, "", "Invalid token", "", "")
+			return
+		} else if parsedToken.Header["alg"] == nil {
+			log.Errorf("alg must be defined")
+			unauthorized(ctx, "", "alg must be defined", "", "")
+			return
+		}
+
+		info = *parsedToken
+		/*err = json.Unmarshal([]byte(parsedToken.Raw), &info)
+		if err != nil {
+			log.Errorf("Error while pasing jwt token : %v.", err)
+			unauthorized(ctx, "", "error parsing jwt token", f.authClient.url.Hostname(), "")
+			return*/
+		//}
+	} else {
+		info = infoTemp.(jwt.Token)
+	}
+
+	/*sub, err := info.Sub()
+	if err != nil {
+		if err != errInvalidTokenintrospectionData {
+			log.Errorf("Error while reading token: %v.", err)
+		}
+
+		unauthorized(ctx, sub, invalidSub, f.authClient.url.Hostname(), "")
+		return
+	}
+
+	if !info.Active() {
+		unauthorized(ctx, sub, inactiveToken, f.authClient.url.Hostname(), "")
+		return
+	}*/
+
+	sub := info.Claims.(jwt.MapClaims)["sub"].(string)
+
+	var allowed = f.validateAnyClaims(info)
+	if !allowed {
+		unauthorized(ctx, sub, invalidClaim, "", "")
+		return
+	}
+
+	authorized(ctx, sub)
+
+	var container2 tokenContainer
+	container2.Claims = make(map[string]interface{})
+	for claim, value := range info.Claims.(jwt.MapClaims) {
+		container2.Claims[claim] = value
+	}
+	//container2.Claims = info.Claims.(jwt.MapClaims)
+
+	log.Errorf("Length %d", len(container2.Claims))
+	ctx.StateBag()[oidcClaimsCacheKey] = container2
+
+	// adding upstream headers
+	f.setHeaders(ctx, info.Claims.(jwt.MapClaims))
+}
+
+func (f *jwtValidationFilter) Response(filters.FilterContext) {}
+
+// Close cleans-up the scheduler
+func (f *jwtValidationFilter) Close() {
+	f.scheduler.Clear()
+}
```

Comment:
This unfortunately would not work as expected due to https://github.com/zalando/skipper/issues/202

Reply:
If I understand properly, you don't want to use scheduler anyways. So, this step will not be necessary then.

Reply:
This is just a note that cleanup unfortunately is not implemented yet though there are filters that implement Close method that is not used